src/compile/model.ts

```diff
@@ -64,6 +70,11 @@ class NameMap {
   }
 }
 
+export interface NameMapShape {
```

Comment:
I'll happily rename it to `NameMapInterface` if you'd prefer. The issue in this case is as follows:
`NameMap` is internal, but protected (and thus, publicly accessible) members of the exported class below reference the type. To avoid needing to export the `NameMap` implementation, I split the interface and the implementation (hence why `NameMap` implements `NameMapInterface`) so I could just export the interface.

An alternate fix is to just export `NameMap`, if you'd prefer.